src/backend/marsha/core/tests/test_services_video_recording.py

```diff
@@ -0,0 +1,155 @@
+"""Tests for the video_recording service in the ``core`` app of the Marsha project."""
+from datetime import timedelta
+from unittest import mock
+
+from django.test import TestCase
+from django.utils import timezone
+
+from ..defaults import HARVESTING, PENDING
+from ..factories import VideoFactory
+from ..services.video_recording import (
+    VideoRecordingError,
+    start_recording,
+    stop_recording,
+)
+from ..utils.time_utils import to_timestamp
+
+
+class VideoModelsTestCase(TestCase):
+    """Test our intentions about the Video model."""
+
+    def test_models_video_start_recording_not_allowed(self):
+        """A video recording cannot be started if not allowed."""
+        video = VideoFactory(allow_recording=False)
+        with self.assertRaises(
+            VideoRecordingError, msg="Video recording is not allowed"
+        ):
+            start_recording(video)
+
+        self.assertEqual(video.recording_slices, [])
+
+    def test_models_video_start_recording_allowed_never_started(self):
+        """A video recording can be started if allowed and has never started."""
+        video = VideoFactory(allow_recording=True)
+        now = timezone.now()
+        with mock.patch.object(timezone, "now", return_value=now):
+            start_recording(video)
+
+        self.assertEqual(video.recording_slices, [{"start": to_timestamp(now)}])
+
+    def test_models_video_start_recording_allowed_not_already_started(self):
+        """A video recording can be started if allowed and not already started."""
+        now = timezone.now()
+        start = now - timedelta(minutes=20)
+        stop = now + timedelta(minutes=10)
```

Comment:
```suggestion
        stop = start + timedelta(minutes=10)
```

Otherwise your next record slice will overlap the next one. And it's not possible with the implementation you made.